client/client.go

```diff
@@ -376,6 +410,8 @@ func (c *Client) ensureRunning() {
 func (c *Client) Send(r *Request) (*amqp.Delivery, error) {
```

Comment:
We could perhaps make it so that we only need one type of middleware instead of `Pre` and `Post`

```go
func (c *Client) Send(r *Request) (*amqp.Delivery, error) {
    c.ensureRunning()
    send := MiddlewarePreChain(c.sendFunc, c.middlewares...)
    return send(r)
}

func (c *Client) sendFunc(r *Request) (*amqp.Delivery, error) {
    // .... Everything currently in Send()
}
```